canvas_grab/__main__.py

```diff
@@ -9,45 +9,92 @@
 import sys
 
 
-def request_reconfigure():
-    if len(sys.argv) >= 2:
-        if sys.argv[1] == 'configure':
-            return True
-    return False
+def get_options():
+    import argparse
 
-
-def main():
-    init()
-
-    # First, welcome our users
-    print("Thank you for using canvas_grab!")
-    print(
-        f'You are using version {canvas_grab.__version__}. If you have any questions, please file an issue at {colored("https://github.com/skyzh/canvas_grab/issues", "blue")}')
-    print(
-        f'You may review {colored("README.md", "green")} and {colored("LICENSE", "green")} shipped with this release')
-    print('--------------------')
-
-    # Then, load config and start setup wizard
-
-    config_file = Path('config.toml')
+    def greeting():
+        # First, welcome our users
+        print("Thank you for using canvas_grab!")
+        print(
+            f'You are using version {canvas_grab.__version__}. If you have any questions, please file an issue at {colored("https://github.com/skyzh/canvas_grab/issues", "blue")}')
+        print(
+            f'You may review {colored("README.md", "green")} and {colored("LICENSE", "green")} shipped with this release')
+        print(
+            f'You may run this code with argument {colored(f"-h","cyan")} for command line usage')
+        print('--------------------')
+
+    # Argument Parser initiation
+    parser = argparse.ArgumentParser(
+        description='Grab all files on Canvas LMS to local directory.')
+
+    # Interactive
+    interactive_group = parser.add_mutually_exclusive_group()
+    interactive_group.add_argument("-i", "--interactive", dest="interactive", action="store_true",
+                                   default=True, help="Set the program to run in interactive mode (default action)")
+    interactive_group.add_argument("-I", "--non-interactive", "--no-input", dest="interactive", action="store_false", default=True,
+                                   help="Set the program to run in non-interactive mode. This can be used to exit immediately in case of profile corruption without getting stuck with the input.")
+
+    # Reconfiguration
+    parser.add_argument("-r", "--reconfigure", "--configure", dest="reconfigure", help="Reconfigure the tool.",
+                        action="store_true")
+
+    # Location Specification
+    parser.add_argument("-o", "--download-folder", "--output",
+                        dest="download", help="Set the download folder.")
```

Comment:
What about using `override download folder` as help here, as `download_folder` is part of the configuration file?

Reply:
I don't think the word "override" should be used in this way. 
I changed it to `Specify alternative download folder.` instead.

Reply:
Alright